framework/src/media/audio/audio_manager.c

```diff
@@ -191,27 +190,26 @@ static audio_manager_result_t find_audio_card(audio_card_type_t card_type)
 			snprintf(card[card_id].card_path, strlen("/dev/audio/") + strlen(dir_entry->d_name) + 1, "%s%s", "/dev/audio/", dir_entry->d_name);
 			card[card_id].status = AUDIO_CARD_IDLE;
 			card[card_id].device_id = device_id;
-			found_cards++;
 			medvdbg("Found an audio card, total card : %d id : %d, card_path : %s\n", found_cards, card_id, card[card_id].card_path);
+
+			/* update actual card here */
+			if (card_type == INPUT) {
```

Comment:
If we don't set  "g_actual_audio_XX_card_id"  here,  this  "find_audio_card()"  is called again in  "get_hardware_params()"...